### ISSUE TITLE
Update telemetrygen span info

### ISSUE THREAD

**Issue by edwintye** [open-telemetry/opentelemetry-collector-contrib#24286]
### Component(s)

cmd/telemetrygen

### Is your feature request related to a problem? Please describe.

Currently telemetrygen adds `span.kind` as an attribute [L43](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/v0.81.0/cmd/telemetrygen/internal/traces/worker.go#L43) but "Span Kind" is neither a resource or an attribute and should be set separately via say `trace.WithSpanKind(trace.SpanKindClient)` instead.

Also for the purpose of testing it may be useful to have the ability to set the `status.code`.

### Describe the solution you'd like

Update telemetrygen in [worker.go](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/v0.81.0/cmd/telemetrygen/internal/traces/worker.go) so something like (at around [L43](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/v0.81.0/cmd/telemetrygen/internal/traces/worker.go#L42))

```go
		ctx, sp := tracer.Start(context.Background(), "lets-go", trace.WithAttributes(
 --                     attribute.String("span.kind", "client"), // is there a semantic convention for this?
			semconv.NetPeerIPKey.String(fakeIP),
			semconv.PeerServiceKey.String("telemetrygen-server"),
		),
++			trace.WithSpanKind(trace.SpanKindClient),
		)
```

and for the server

```go
		_, child := tracer.Start(childCtx, "okey-dokey", trace.WithAttributes(
--                      attribute.String("span.kind", "server"),
			semconv.NetPeerIPKey.String(fakeIP),
			semconv.PeerServiceKey.String("telemetrygen-client"),
		),
++			trace.WithSpanKind(trace.SpanKindServer),
		)
```

Then maybe hard code the status code at [L68](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/v0.81.0/cmd/telemetrygen/internal/traces/worker.go#L68) 

```go
		opt := trace.WithTimestamp(time.Now().Add(fakeSpanDuration))
++		child.SetStatus(codes.Ok, "Success")
		child.End(opt)
++		sp.SetStatus(codes.Ok, "Success")
		sp.End(opt)
```

or some alternative such as allow user inject via command line argument.

### Additional context

This can be easily verified using a rudimentary config locally

```yaml
receivers:
  otlp:
    protocols:
      grpc:
      http:
exporters:
  logging:
    verbosity: detailed
service:
  pipelines:
    traces/raw:
      receivers: [otlp]
      processors: []
      exporters: [logging]
```

and running telemetrygen against it via

```shell
telemetrygen traces --otlp-insecure --traces 1
```

to confirm that telemetrygen currently produces

```
esource SchemaURL: https://opentelemetry.io/schemas/1.4.0
Resource attributes:
     -> service.name: Str(telemetrygen)
ScopeSpans #0
ScopeSpans SchemaURL:
InstrumentationScope telemetrygen
Span #0
    Trace ID       : 7369a96e4f44cec645ad55e0637e136f
    Parent ID      : 70692ee1b35b1411
    ID             : 05a70278c3c4fa07
    Name           : okey-dokey
    Kind           : Internal
    Start time     : 2023-07-16 18:53:34.178838 +0000 UTC
    End time       : 2023-07-16 18:53:34.178962 +0000 UTC
    Status code    : Unset
    Status message :
Attributes:
     -> span.kind: Str(server)
     -> net.peer.ip: Str(1.2.3.4)
     -> peer.service: Str(telemetrygen-client)
```

which has `Kind: Internal` and `span.kind: "server"` without a status code set.  This caused slight confusion when we were trying to test out some tail sampling policy locally.

**Comment by mx-psi** [comment]
Setting span kind with `WithSpanKind` makes sense to me. We can open a separate PR to discuss the status code setting.

**Comment by crobert-1** [comment]
Hello @edwintye, was this issue resolved with #24303, or is there still some work required? If it's complete we can close the issue.
